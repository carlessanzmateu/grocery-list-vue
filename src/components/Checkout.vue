<template>
  <section class="checkout w-full p-5">
    <p class="total-amount pb-5">Total Amount: {{ totalAmount }}$</p>
    <a class="button w-full p-2">Checkout</a>
  </section>
</template>

<script>
export default {
  name: 'Checkout',
  props: {
    totalAmount: {
      type: Number,
      required: true
    }
  }
}
</script>

<style lang="scss" scoped>
.checkout {
  .button {
    height: 40px;
    color: var(--white-500);
    background-color: var(--blue-500);
    line-height: 37px;

    &:hover {
      background-color: var(--violet-500);
    }
  }
}
</style>
