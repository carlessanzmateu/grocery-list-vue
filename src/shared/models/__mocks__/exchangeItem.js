const mockExchangeItem = {
  id: '1',
  // eslint-disable-next-line
  image_url: 'mock.png',
  stock: 1,
  productName: 'some name',
  price: 10,
  productDescription: 'Some description',
  favorite: 0
}
