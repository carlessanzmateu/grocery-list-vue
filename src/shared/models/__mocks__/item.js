class MockItem {
  constructor() {
    this.$id = '1'
    this.$image = 'mock.png'
    this.$stockReadyToBuy = 0
    this.$stock = 1
    this.$name = 'some name'
    this.$price = 10
    this.$description = 'Some description'
    this.$favorite = false
  }

  getId () {
    return this.$id
  }

  getImage () {
    return this.$image
  }

  initSelectedStock () {
    this.$stockReadyToBuy = 0
  }

  getSelectedStock () {
    return this.$stockReadyToBuy
  }

  getStock () {
    return this.$stock
  }

  getStockLeft () {
    return this.getStock() - this.getSelectedStock()
  }

  canIncreaseSelectedStock () {
    return this.getSelectedStock() < this.getStock()
  }

  canDecreaseSelectedStock () {
    return this.getSelectedStock() > 0
  }

  increaseSelectedStock () {
    if (!this.canIncreaseSelectedStock()) {
      return
    }

    this.$stockReadyToBuy++
  }

  decreaseSelectedStock () {
    if (!this.canDecreaseSelectedStock()) {
      return
    }

    this.$stockReadyToBuy--
  }

  getName () {
    return this.$name
  }

  getPrice () {
    return this.$price
  }

  getItemTotalPrice () {
    return this.getSelectedStock() * this.getPrice()
  }

  getDescription () {
    return this.$description
  }

  buildFavorite (favorite) {
    this.$favorite = (favorite === '1')
  }

  getIsFavorite () {
    return this.$favorite
  }

  setAsFavorite () {
    this.$favorite = true
  }

  removeFromFavorite () {
    this.$favorite = false
  }

  toggleFavorite () {
    this.$favorite = !this.$favorite
  }
}

export { MockItem }
