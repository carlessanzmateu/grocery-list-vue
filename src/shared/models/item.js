export default class Item {
  $id
  $image
  $stockReadyToBuy
  $stock
  $name
  $price
  $description
  $favorite

  constructor (rawItem) {
    this.$id = rawItem.id
    this.$image = rawItem.image_url
    this.$stock = rawItem.stock
    this.$name = rawItem.productName
    this.$price = rawItem.price
    this.$description = rawItem.productDescription
    this.buildFavorite(rawItem.favorite)
    this.initSelectedStock()
  }

  getId () {
    return this.$id
  }

  getImage () {
    return this.$image
  }

  initSelectedStock () {
    this.$stockReadyToBuy = 0
  }

  getSelectedStock () {
    return this.$stockReadyToBuy
  }

  // setSelectedStock (selectedStock) {
  //   this.$stockReadyToBuy = selectedStock
  // }

  getStock () {
    return this.$stock
  }

  setStock (stock) {
    this.$stock = stock
  }

  getStockLeft () {
    return this.getStock() - this.getSelectedStock()
  }

  canIncreaseSelectedStock () {
    return this.getSelectedStock() < this.getStock()
  }

  canDecreaseSelectedStock () {
    return this.getSelectedStock() > 0
  }

  increaseSelectedStock () {
    if (!this.canIncreaseSelectedStock()) {
      return
    }

    this.$stockReadyToBuy++
  }

  decreaseSelectedStock () {
    if (!this.canDecreaseSelectedStock()) {
      return
    }

    this.$stockReadyToBuy--
  }

  getName () {
    return this.$name
  }

  getPrice () {
    return this.$price
  }

  getItemTotalPrice () {
    return this.getSelectedStock() * this.getPrice()
  }

  getDescription () {
    return this.$description
  }

  buildFavorite (favorite) {
    this.$favorite = (favorite === '1')
  }

  getIsFavorite () {
    return this.$favorite
  }

  setAsFavorite () {
    this.$favorite = true
  }

  removeFromFavorite () {
    this.$favorite = false
  }

  toggleFavorite () {
    this.$favorite = !this.$favorite
  }
}
