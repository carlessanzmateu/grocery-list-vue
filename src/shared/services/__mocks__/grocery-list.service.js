export default {
  getAllGroceries: jest.fn(),
  getAllFavorites: jest.fn(),
  updateGrocery: jest.fn()
}
