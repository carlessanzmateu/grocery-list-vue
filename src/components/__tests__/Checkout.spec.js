import { shallowMount } from '@vue/test-utils'
import Checkout from '@/components/Checkout.vue'
import { MockItem } from '@/shared/models/__mocks__/item'

describe('Checkout', () => {
  let wrapper
  function createWrapper() {
    return shallowMount(Checkout)
  }

  it('should create', () => {
    wrapper = createWrapper()
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('should match html', () => {
    wrapper = createWrapper()
    expect(wrapper.element).toMatchSnapshot()
  })
})
