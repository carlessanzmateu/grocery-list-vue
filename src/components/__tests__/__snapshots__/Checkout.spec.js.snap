// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout should match html 1`] = `
<section
  class="checkout w-full p-5"
>
  <p
    class="total-amount pb-5"
  >
    Total Amount: 10$
  </p>
   
  <a
    class="button w-full p-2"
  >
    Checkout
  </a>
</section>
`;
